webpack/scenes/ContentViews/Publish/PublishContentViewWizard.js

```diff
@@ -0,0 +1,130 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { STATUS } from 'foremanReact/constants';
+import { Wizard } from '@patternfly/react-core';
+import { translate as __ } from 'foremanReact/common/I18n';
+import CVPublishForm from './CVPublishForm';
+import CVPublishFinish from './CVPublishFinish';
+import getEnvironmentPaths from '../components/EnvironmentPaths/EnvironmentPathActions';
+import CVPublishReview from './CVPublishReview';
+import {
+  selectEnvironmentPaths,
+  selectEnvironmentPathsStatus,
+} from '../components/EnvironmentPaths/EnvironmentPathSelectors';
+
+const PublishContentViewWizard = ({
+  details, show, setIsOpen, currentStep, setCurrentStep,
+}) => {
+  const { name, id: cvId, version_count: versionCount } = details;
+  const [description, setDescription] = useState('');
+  // Change to userCheckedItems
+  const [userCheckedItems, setUserCheckedItems] = useState([]);
+  const [promote, setPromote] = useState(false);
+  const [forcePromote, setForcePromote] = useState([]);
+  const [envPathFlat, setEnvPathFlat] = useState([]);
+  const dispatch = useDispatch();
+  const environmentPathResponse = useSelector(selectEnvironmentPaths);
+  const environmentPathStatus = useSelector(selectEnvironmentPathsStatus);
+  const environmentPathLoading = environmentPathStatus === STATUS.PENDING;
+
+  const steps = [
+    {
+      id: 1,
+      name: 'Publish',
+      component: <CVPublishForm
+        description={description}
+        setDescription={setDescription}
+        details={details}
+        userCheckedItems={userCheckedItems}
+        setUserCheckedItems={setUserCheckedItems}
+        promote={promote}
+        setPromote={setPromote}
+        forcePromote={forcePromote}
+      />,
+    },
+    {
+      id: 2, name: 'Review', component: <CVPublishReview details={details} userCheckedItems={userCheckedItems} />, nextButtonText: 'Finish',
+    },
+    {
+      id: 3,
+      name: 'Finish',
+      component: <CVPublishFinish
+        description={description}
+        setDescription={setDescription}
+        userCheckedItems={userCheckedItems}
+        setUserCheckedItems={setUserCheckedItems}
+        forcePromote={forcePromote}
+        cvId={cvId}
+        versionCount={versionCount}
+        show={show}
+        setIsOpen={setIsOpen}
+        setCurrentStep={setCurrentStep}
+      />,
+      isFinishedStep: true,
+    },
+  ];
+
+  useEffect(
+    () => {
+      dispatch(getEnvironmentPaths());
+      setCurrentStep(1);
+    },
+    [],
```

Comment:
dependency array needs to be filled out

Reply:
This is intentional. This is supposed to run just once on component load.

Reply:
eslint isn't going to care about that once we turn on that rule ;)

Reply:
https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies

Reply:
Oh, and if you want to read the whole novel about it: 
https://github.com/facebook/react/issues/14920

Reply:
They do discuss using props and functions with side effects on props in dependency array. Added the functions to dependency array..